data/txHandler.go

```diff
@@ -282,41 +309,135 @@ func (handler *TxHandler) asyncVerifySignature(arg interface{}) interface{} {
 		// we failed to write to the output queue, since the queue was full.
 		// adding the metric here allows us to monitor how frequently it happens.
 		transactionMessagesDroppedFromPool.Inc(nil)
+
+		// delete from duplicate caches to give it chance to be re-submitted
+		// this relatively rare operation and implementation is expensive (requires re-hashing)
+		handler.deleteFromCaches(tx.rawmsg.Data, tx.unverifiedTxGroup)
```

Comment:
you are doing all these work without checking `handler.cacheConfig.enableFilteringCanonical`